File: Erik_Vincent/Assignment1/Assignment1/Model/QLS/Widget.cs
using System.Collections.Generic;

namespace Assignment1.Model.QLS
{
    public abstract class Widget
    {
    }

    public class CheckBoxWidget : Widget { }

    public class RadioWidget : Widget
    {
        public readonly string Yes, No;
        public RadioWidget(string yes, string no)
        {
            Yes = yes;
            No = no;
        }
    }

    public class SliderWidget : Widget { }
    public class SpinBoxWidget : Widget { }
    public class TextBoxWidget : Widget { }

    public class DropDownWidget : Widget
    {
        public readonly string Yes, No;
        public DropDownWidget(string yes, string no)
        {
            Yes = yes;
            No = no;
        }
    }
}

File: Remi_Hasan/src/main/java/qls/model/widget/Widget.java
package qls.model.widget;

import qls.model.QLSNode;

public abstract class Widget extends QLSNode {
    private final WidgetType type;

    Widget(WidgetType type) {
        this.type = type;
    }

    public WidgetType getType() {
        return type;
    }
}